app/models/partners/family.rb

```diff
@@ -28,6 +29,7 @@ module Partners
   class Family < Base
     belongs_to :partner, class_name: '::Partner'
     has_many :children, dependent: :destroy
+    before_save :archive_children, if: :archived?
```

Comment:
I don't think this should be in a callback. Better to make this a service object and archive the family and children together. Callbacks aren't ideal to work with associations.

Reply:
Thank you for the feedback! I will work on implementing that

Reply:
I took a crack at moving it to a service object, let me know what you think!